sample-app/src/hooks/usePage.tsx

```diff
@@ -0,0 +1,33 @@
+import { SearchIntent } from "@yext/answers-headless";
+import { useLayoutEffect } from "react";
+import { useAnswersActions } from "@yext/answers-headless-react";
+import { executeSearch, getSearchIntents, updateLocationIfNeeded } from "../utils/search-operations";
+
+/**
+ * Sets up the state for a page
+ * @param verticalKey - The verticalKey associated with the page, or undefined for universal pages
+ */
+export default function usePage(verticalKey?: string) {
```

Comment:
Nit: Could we call this `setupPage` or something like that? The `use` prefix suggests a hook which will return a value and a setter.

Reply:
I think this needs to have a use prefix to be consider a custom hook so it can call other hooks (e.g. useAnswersActions, useLayoutEffect), otherwise react would yell at you.

Reply:
I like 'useAnswersOnPage'